spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/batch/BatchConfigurerConfiguration.java

```diff
@@ -37,10 +37,10 @@
 @ConditionalOnClass(PlatformTransactionManager.class)
 @ConditionalOnMissingBean(BatchConfigurer.class)
 @Configuration(proxyBeanMethods = false)
+@ConditionalOnBean(DataSource.class)
 class BatchConfigurerConfiguration {
 
 	@Configuration(proxyBeanMethods = false)
-	@ConditionalOnBean(DataSource.class)
```

Comment:
This one highlights another bug that's similar to https://github.com/spring-projects/spring-boot/issues/17449. In the (very) unlikely event of there being an `entityManagerFactory` bean that has been defined without exposing a `DataSource` as a bean, I believe the auto-configuration will fail. We should fix this in 2.1.x.